### PR TITLE
feat(ui): Button `busy` states

### PR DESCRIPTION
**Pull request by joe-bell**
## Description

Adds a `busy` state to both the `Connection.Button` and `Button`

I intentionally chose to call this `busy` to create a clearer separation from "data loading" (e.g. page loading where placeholders might be needed)

The idea here is that Clerk is "busy" processing a user action

I haven't added any animations/motion yet; that's something we'll need to come back to

Additionally, I've increased the opacity reduction of `Connection.Button` when `disabled` to put more prominence on the `busy` `Connection.Button`

https://github.com/clerk/javascript/assets/7349341/65d8bd86-dc16-42ca-b6bf-1dcfba0c7ca9

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/ui/src/global.css [hunk 1]
```diff
@@ -1,7 +1,17 @@
+@keyframes cl-spin {
+  to {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
+
```
Comment (joe-bell) [inline packages/ui/src/global.css:9]: Shared privately; unfortunately we can't use the included `@keyframes` or the `keyframes` key in `tailwind.config.ts`

**Review by alexcarpenter — APPROVED** [review]

File: packages/ui/src/primitives/spinner.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    className,
+  }: Pick<HTMLAttributes<SpinnerRef>, 'className'> & {
+    children: string;
+  },
+  ref: React.ForwardedRef<SpinnerRef>,
+) {
+  return (
+    <span
+      ref={ref}
+      className={cn('relative isolate text-current size-[1em] block', className)}
```
Comment (alexcarpenter) [inline packages/ui/src/primitives/spinner.tsx:39]: @joe-bell added block here as the size value isnt respected for inline elements.

File: package-lock.json [hunk 3]
```diff
@@ -38920,18 +38920,18 @@
       "devDependencies": {
         "@clerk/eslint-config-custom": "*",
         "@cloudflare/workers-types": "^3.18.0",
-        "@types/chai": "^4.3.16",
+        "@types/chai": "^4.3.3",
```
Comment (alexcarpenter) [inline package-lock.json:38923]: @BRKalow can you review that these changes are ok since it seems unexpected for these changes to be included in this pr?

Comment (joe-bell) [reply package-lock.json:38923]: Weird, no idea why this happened

Comment (alexcarpenter) [reply package-lock.json:38923]: this is happening on main as well, nothing done in this pr

Comment (alexcarpenter) [reply package-lock.json:38923]: Renovate update. Got confirmation this is good to go.

**Pull request merged**